lib/OpenQA/Resource/Locks.pm

```diff
@@ -126,18 +116,14 @@ sub barrier_wait ($name, $jobid, $where, $check_dead_job) {
         }
     }
 
-    if (grep { $_ eq $jobid } @jobs) {
-        return 1 if @jobs == $barrier->count;
-        return 0;
-    }
+    return @jobs == $barrier->count if grep { $_ eq $jobid } @jobs;
```

Comment:
The code invoking this function actually does a numerical comparison with the return value. We shouldn't just return a boolean. The same counts for the other occurrences.

Reply:
A boolean created like this will already behave like expected in numeric context without a warning.
```
$b = 23 == 24; # magic boolean that just looks like an empty string
$s = ""; # just an empty string
say "\$b:";
say $b + 1; 
say "\$s:";
say $s + 1;
__END__
$b:
1
$s:
Argument "" isn't numeric in addition (+) at -e line 7.
1
```